ui/desktop/src/components/settings/modals/AddExtensionDialog.tsx

```diff
@@ -1,64 +1,104 @@
 import React from 'react';
-import { BaseDialog } from './BaseDialog';
-import { Extension } from '../types';
-import { Input } from '../../ui/input';
+import { Modal, ModalContent, ModalHeader, ModalTitle } from '../../ui/modal';
+import { Button } from '../../ui/button';
+import { Extension, EnvironmentVariable } from '../types';
 
 interface AddExtensionDialogProps {
     isOpen: boolean;
     onClose: () => void;
-    onAdd: (extension: Extension) => void;
+    extension: Extension | null;
+    onSubmit: (envVars: Record<string, string>) => void;
 }
 
-export function AddExtensionDialog({ isOpen, onClose, onAdd }: AddExtensionDialogProps) {
-    const [name, setName] = React.useState('');
-    const [description, setDescription] = React.useState('Standard config');
+export function AddExtensionDialog({
+    isOpen,
+    onClose,
+    extension,
+    onSubmit,
+}: AddExtensionDialogProps) {
+    const [envValues, setEnvValues] = React.useState<Record<string, string>>({});
 
-    // Reset form when dialog closes
+    // Reset form when dialog closes or extension changes
     React.useEffect(() => {
-        if (!isOpen) {
-            setName('');
-            setDescription('Standard config');
+        if (!isOpen || !extension) {
+            setEnvValues({});
         }
-    }, [isOpen]);
+    }, [isOpen, extension]);
 
     const handleSubmit = (e: React.FormEvent) => {
         e.preventDefault();
-        const id = name.toLowerCase().replace(/\s+/g, '-');
-        onAdd({
-            id,
-            name,
-            description,
-            enabled: true
-        });
+        onSubmit(envValues);
         onClose();
     };
 
+    if (!extension) return null;
+
+    const requiredEnvVars = extension.environmentVariables.filter(v => v.required);
+
     return (
-        <BaseDialog 
-            title="Add Extension" 
-            isOpen={isOpen} 
-            onClose={onClose}
-            onSubmit={handleSubmit}
-        >
-            <form id="keyForm" onSubmit={handleSubmit} className="space-y-4">
-                <div>
-                    <label className="text-sm font-medium">Name</label>
-                    <Input
-                        value={name}
-                        onChange={(e) => setName(e.target.value)}
-                        placeholder="Extension name"
-                        required
-                    />
-                </div>
-                <div>
-                    <label className="text-sm font-medium">Description</label>
-                    <Input
-                        value={description}
-                        onChange={(e) => setDescription(e.target.value)}
-                        placeholder="Extension description"
-                    />
-                </div>
-            </form>
-        </BaseDialog>
+        <Modal open={isOpen} onOpenChange={onClose}>
```

Comment:
I wasn't able to get this to open up as is, but hacked around it. this would need to be activated from the DEEP_LINK when creating the `BrowserWindow` in main.ts - so that it ends up showing this with the values from it. I think it could possibly also have a link people can click on which lets you type in command etc (less clear about that - and need to resolve if store command or cmd + args first). Then on save it can apply them to current window